semantic-python/src/Language/Python/Core.hs

```diff
@@ -115,12 +111,10 @@ locate syn item = do
 defaultCompile :: (MonadFail m, Show py) => py -> m (t Name)
 defaultCompile t = fail $ "compilation unimplemented for " <> show t
 
-newtype CompileSum py a = CompileSum (py a)
 
-instance (Generic1 py, GCompileSum (Rep1 py)) => Compile (CompileSum py) where
-  compileCC (CompileSum a) cc = gcompileCCSum (from1 a) cc
-
-deriving via CompileSum (l :+: r) instance (Compile l, Compile r) => Compile (l :+: r)
+instance (Compile l, Compile r) => Compile (l :+: r) where
+  compileCC (L1 l) cc = compileCC l cc
+  compileCC (R1 r) cc = compileCC r cc
```

Comment:
This instance does all of the legwork for sums both named and anonymous.

Reply:
I absolutely love this, and sincerely regret doubting the merit of the move from named to anonymous sums.

Reply:
[![…and you doubted me, William.](https://user-images.githubusercontent.com/59671/66432548-576fa900-e9ec-11e9-9939-dc98c461dee3.png)](https://pbfcomics.com/comics/pyro-billy/)